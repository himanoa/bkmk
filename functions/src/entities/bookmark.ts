interface Bookmark {
  id: string;
  url: string;
  bodyHtml: string;
  comment: string;
}

