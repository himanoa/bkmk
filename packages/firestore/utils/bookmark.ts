import * as faker from "faker"
import * as firebase from "@firebase/testing"

export default function(overwrite?) {
  return {
    url: faker.internet.url(),
    bodyHtml: faker.lorem.text(),
    bodyText: faker.lorem.text(),
    title: faker.lorem.sentence(),
    comment: faker.lorem.paragraph(),
    createdAt: firebase.firestore.Timestamp.now(),
    updatedAt: firebase.firestore.Timestamp.now()
  }
}
