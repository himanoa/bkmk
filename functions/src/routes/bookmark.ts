import * as Express from "express";
import { parseInput } from "../entities/bookmark"
import * as admin from 'firebase-admin';

const router = Express.Router();

router.post("/", async (req, res) => {
  const bookmark = parseInput(req.body)
  console.dir(await admin.firestore().collection('bookmark').add(bookmark))
})


export default {
  path: "/bookmarks",
  router
};
