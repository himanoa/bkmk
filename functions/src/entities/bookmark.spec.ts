import { parse } from "./bookmark"

describe("parse", () => {
  it("should be valid", () => {
    const target = {
      id: "valid",
      url: "https://example.com",
      bodyHtml: "<html></html>",
      comment: "hello"
    }
    const actual = parse(target)
    expect(actual).toBe(target)
  })
  it("should not be valid when invalid url", () => {
    const target = {
      id: "valid",
      url: "example",
      bodyHtml: "<html></html>",
      comment: "hello"
    }
    expect(() => parse(target)).toThrowErrorMatchingSnapshot()
  })
})

