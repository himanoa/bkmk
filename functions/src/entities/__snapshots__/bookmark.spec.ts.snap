// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse should not be valid when invalid url 1`] = `"url must be a valid URL"`;
